### PR TITLE
Add Hall sensor support

### PR DESCRIPTION
**Pull request by BluCodeGH**
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/cansw_actuator/11)

### CODE REVIEW

File: actuator.c [hunk 1]
```diff
@@ -47,6 +53,14 @@ void actuator_set(enum ACTUATOR_STATE state) {
...
 #if !HAS_LIMS
     return ACTUATOR_UNK;
+#else
+// define in board.h
+#ifdef INJECTOR
+    adc_result_t hall_raw = ADCC_GetSingleConversion(channel_HALL);
+    
+    if (hall_raw > HALL_ERR_THRESHOLD) { return ACTUATOR_ILLEGAL; }
+    if (hall_raw > HALL_THRESHOLD) { return HIGH_STATE; }
+    return 1 - HIGH_STATE;
```
Comment (Rio6) [inline actuator.c]: why is there no LOW_STATE?

Comment (BluCodeGH) [reply actuator.c]: HIGH_STATE is a macro defined in board.h that maps a high hall sensor reading to an ACTUATOR_STATE. This code is confusing tho, I'll add a comment and explicit cast.

Comment (Rio6) [reply actuator.c]: Can just `#define LOW_STATE (1 - HIGH_STATE)` right below where the HIGH_STATE is defined

Comment (BluCodeGH) [reply actuator.c]: This code is all getting rewritten with propulsion board so I'd rather not worry about it

Comment (Rio6) [reply actuator.c]: ok

**Review by Rio6 — APPROVED** [review]

**Pull request merged**